R/utils.R

```diff
@@ -19,43 +19,47 @@ warnf <- function(fmt, ..., call. = TRUE)
   ))
 }
 
-# TODO: Windows
 copy_directory <- function(source,
                            target,
                            overwrite = TRUE,
-                           exclude = character())
-{
-  # validate source, target directories
+                           exclude = character(),
+                           include = character()) {
+
+  # source dir
   source <- normalizePath(source, winslash = "/", mustWork = TRUE)
 
+  # target dir
   if (file.exists(target)) {
     if (!overwrite)
       stopf("a file already exists at path '%s'", target)
     unlink(target, recursive = TRUE)
   }
-
-  # construct exclusion string for rsync
-  shexclude <- if (length(exclude))
-    sprintf("--exclude={%s}", paste(exclude, collapse = ","))
-
-  # call rsync to copy directory contents to new directory
-  # trailing slash on 'source' lets us override dirname
-  system(paste(
-    "rsync -a",
-    shQuote(paste(source, "/", sep = "")),
-    shQuote(target),
-    if (length(shexclude)) shexclude
-  ))
-
-  isdir <- isTRUE(file.info(target)$isdir)
-  if (!isdir) {
-    fmt <- "failed to copy '%s' => '%s': unknown reason"
-    stopf(fmt, source, target)
+  dir.create(target)
+
+  # get the original top level file listing
+  all_files <- list.files(source, all.files = TRUE)
+
+  # apply excludes to the top level listing
+  exclude <- c("^\\..*$", utils::glob2rx(exclude))
```

Comment:
Actually, with that I was just trying to exclude `.` and `..` so I think I brought over the wrong regex! I'll fix that.